src/v/storage/segment.h

```diff
@@ -163,6 +163,11 @@ class segment {
     generation_id get_generation_id() const { return _generation_id; }
     void advance_generation() { _generation_id++; }
 
+    auto first_write_ts()
+      -> std::optional<std::chrono::system_clock::time_point> {
```

Comment:
nit: as stylish as this is, it's probably more immediately recognizable to write this without `auto`